frontend/packages/pipelines-plugin/src/components/import/pipeline/pipeline-template-utils.ts

```diff
@@ -79,3 +106,44 @@ export const createPipelineRunForImportFlow = async (pipeline: Pipeline): Promis
   };
   return submitStartPipeline(pipelineInitialValues, pipeline);
 };
+
+export const updatePipelineForImportFlow = async (
+  pipelineData: PipelineData,
+  name: string,
+  namespace: string,
+  gitUrl: string,
+  gitRef: string,
+  gitDir: string,
+  dockerfilePath: string,
+): Promise<Pipeline> => {
+  const { data: pipeline, template } = pipelineData;
+  let updatedPipeline = _.cloneDeep(pipeline);
+
+  if (!template) {
+    updatedPipeline.metadata.labels = _.omit(
+      updatedPipeline.metadata.labels,
+      'app.kubernetes.io/instance',
+    );
+  } else {
+    const runtimeLabel = 'pipeline.openshift.io/runtime';
+    if (template.metadata?.labels[runtimeLabel] !== pipeline.metadata?.labels[runtimeLabel]) {
+      updatedPipeline = _.cloneDeep(template) as Pipeline;
+      updatedPipeline.metadata = {
+        name: `${name}`,
+        namespace,
+        labels: { ...template.metadata?.labels, 'app.kubernetes.io/instance': name },
+      };
```

Comment:
@rottencandy when you are updating the `pipeline` with the new `template` you will need to add the `resourceVersion` of the `pipeline` in `updatedPipeline` otherwise the update request fails with the below error:
` Error "Invalid value: 0x0: must be specified for an update" for field "metadata.resourceVersion".`
```suggestion
      updatedPipeline.metadata = {
        resourceVersion: pipeline.metadata.resourceVersion,
        name: `${name}`,
        namespace,
        labels: { ...template.metadata?.labels, 'app.kubernetes.io/instance': name },
      };
```

Reply:
Updated.